contracts/cw-proposal-single/src/contract.rs

```diff
@@ -134,8 +138,28 @@ pub fn execute_propose(
     title: String,
     description: String,
     msgs: Vec<CosmosMsg<Empty>>,
+    proposer: Option<String>,
 ) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
+    let proposal_creation_policy = CREATION_POLICY.load(deps.storage)?;
+
+    // Check that the sender is permitted to create proposals.
+    if !proposal_creation_policy.is_permitted(&sender) {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Determine the appropriate proposer. If this is coming from our
+    // pre-propose module, it must be specified. Otherwise, the
+    // proposer should not be specified.
+    let proposer = match (proposer, proposal_creation_policy) {
+        (None, ProposalCreationPolicy::Anyone {}) => sender.clone(),
```

Comment:
so say the prepropose module fails open and we default to ``ProposalCreationPolicy::Anyone`, allowing anyone to potentially create spam proposals. Wouldn't we instead want to default to allowing only members to submit proposals as before?

Reply:
after discussion, we will leave as is -- this is a failure stopgap and if the module ends up in that state we have bigger issues